zappa/core.py

```diff
@@ -1483,34 +1486,18 @@ def rollback_lambda_function_version(
 
         return response["FunctionArn"]
 
-    def is_lambda_function_ready(self, function_name):
-        """
-        Checks if a lambda function is active and no updates are in progress.
-        """
-        response = self.lambda_client.get_function(FunctionName=function_name)
-        return (
-            response["Configuration"]["State"] == "Active"
-            and response["Configuration"]["LastUpdateStatus"] != "InProgress"
-        )
-
-    def wait_until_lambda_function_is_ready(self, function_name):
+    def wait_until_lambda_function_is_ready(self, lambda_arn):
         """
-        Continuously check if a lambda function is active.
-        For functions deployed with a docker image instead of a
-        ZIP package, the function can take a few seconds longer
-        to be created or update, so we must wait before running any status
-        checks against the function.
+        Wait until lambda State=Active and LastUpdateStatus=Successful
         """
-        show_waiting_message = True
-        while True:
-            if self.is_lambda_function_ready(function_name):
-                break
-
-            if show_waiting_message:
-                print("Waiting until lambda function is ready.")
-                show_waiting_message = False
-
-            time.sleep(1)
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#waiters
+        kwargs = dict(FunctionName="{}:{}".format(lambda_arn, ALB_LAMBDA_ALIAS))
+        waiter = self.lambda_client.get_waiter("function_active")
+        print(f"Waiting for lambda function [{lambda_arn}] to become active...")
+        waiter.wait(**kwargs)
+        waiter = self.lambda_client.get_waiter("function_updated")
```

Comment:
From my reading of the [AWS docs](https://docs.aws.amazon.com/lambda/latest/dg/functions-states.html), it looks like when creating a function, we'd only need to check the `function_active` waiter and when updating a function, we'd only need to check the `function_updated` waiter.  Would it make sense then to split these up so that we are only doing the necessary check in each case rather than always doing both?